src/plugins/discover/public/build_services.ts

```diff
@@ -123,6 +124,7 @@ export function buildServices(
     toastNotifications: core.notifications.toasts,
     uiSettings: core.uiSettings,
     visualizations: plugins.visualizations,
+    useNewQuerySelector: !!plugins.observabilityDashboards,
```

Comment:
Shall we leverage `capabilities` service to check if `PPL` is enabled instead of relying on the existence of observabilityDashboards plugin?

Reply:
does observability dashboards populate the capabilities service?

ideally, i think the best route would be a public registry, (if it's not available already) for the query language. for which plugins can populate with their language and routes like how lucene and dql does. so when a plugin starts up it registers its query language and the routes it should hit.

however, i think this is done in its current state due to priorities in external plugins and that the observability plugin is suppose to support the functionality of re-routing. in attempt to avoid confusion where i can go from [observability data explorer](https://playground.opensearch.org/app/observability-logs#/explorer/Zr-XIowB4v5SU7kSv17P) to [discover data explorer ](https://playground.opensearch.org/app/data-explorer/discover#?_a=(discover:(columns:!(_source),isDirty:!f,sort:!()),metadata:(indexPattern:d3d7af60-4c81-11e8-b3d7-01146121b73d,view:discover))&_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-15h,to:now))&_q=(filters:!(),query:(language:kuery,query:''))). Where when the user selects DQL it takes you to the Discover view (even though there appears to be a bug that prevents that feature.

cc: @abbyhu2000 @ashwin-pc is that correct?

Reply:
Yea when user selects DQL on observability page, it will take user to data explorer. 

Also updated to use application service to determine if observability plugin is installed.